Tests/test_content.py

```diff
@@ -52,53 +58,153 @@ def options_handler():
                                                       'tests on(Valid only when using AMI)', default="NonAMI")
 
     options = parser.parse_args()
+    tests_settings = TestsSettings(options)
+    return tests_settings
+
+
+class TestsSettings:
+    def __init__(self, options):
+        self.api_key = options.apiKey
+        self.server = options.server
+        self.conf_path = options.conf
+        self.secret_conf_path = options.secret
+        self.nightly = options.nightly
+        self.slack = options.slack
+        self.circleci = options.circleci
+        self.buildNumber = options.buildNumber
+        self.buildName = options.buildName
+        self.isAMI = options.isAMI
+        self.memCheck = options.memCheck
+        self.serverVersion = options.serverVersion
+        self.serverNumericVersion = None
+
+
+class PrintJob:
+    def __init__(self, message_to_print, print_function_to_execute, message_color=None):
+        self.print_function_to_execute = print_function_to_execute
+        self.message_to_print = message_to_print
+        self.message_color = message_color
+
+    def execute_print(self):
+        if self.message_color:
+            self.print_function_to_execute(self.message_to_print, self.message_color)
+        else:
+            self.print_function_to_execute(self.message_to_print)
+
 
-    return options
+class ParallelPrintsManager:
 
+    def __init__(self, number_of_threads):
+        self.threads_print_jobs = [[] for i in range(number_of_threads)]
+        self.print_lock = threading.Lock()
+
+    def add_print_job(self, message_to_print, print_function_to_execute, thread_index, message_color=None):
+        if print_color:
+            print_job = PrintJob(message_to_print, print_function_to_execute, message_color)
+        else:
+            print_job = PrintJob(message_to_print, print_function_to_execute)
+        self.threads_print_jobs[thread_index].append(print_job)
+
+    def execute_thread_prints(self, thread_index):
+        self.print_lock.acquire()
+        prints_to_execute = self.threads_print_jobs[thread_index]
+        for print_job in prints_to_execute:
+            print_job.execute_print()
+        self.print_lock.release()
+        self.threads_print_jobs[thread_index] = []
+
+
+class TestsDataKeeper:
+
+    def __init__(self):
+        self.succeeded_playbooks = []
+        self.failed_playbooks = []
+        self.skipped_tests = []
+        self.skipped_integrations = []
+        self.rerecorded_tests = []
+        self.empty_files = []
+        self.unmockable_integrations = {}
```

Comment:
you can use those directly in your code and then you won't need the add_tests_data function